ibis/backends/flink/registry.py

```diff
@@ -207,6 +267,58 @@ def _floor_divide(translator: ExprTranslator, op: ops.Node) -> str:
     return f"FLOOR(({left}) / ({right}))"
 
 
+def _array_index(translator: ExprTranslator, op: ops.arrays.ArrayIndex):
+    table_column = op.arg
+    index = op.index
+
+    table_column_translated = translator.translate(table_column)
+    index_translated = translator.translate(index)
+
+    return f"{table_column_translated} '[' {index_translated} ']'"
+
+
+def _day_of_week_index(translator: ExprTranslator, op: ops.Node) -> str:
+    arg = translator.translate(op.args[0])
+    return f"MOD(DAYOFWEEK({arg}) + 5, 7)"
```

Comment:
Nice.

Nit: Nothing to change, but I just wonder why it wouldn't make sense to use this as the default SQL translation for this operation. Adding 5 just seems like it has to always work, compared to -2 that can result in complications with negative numbers.

Reply:
> Adding 5 just seems like it has to always work, compared to -2 that can result in complications with negative numbers.

I agree. Let's enumerate all possibilities.

> DAYOFWEEK(date) :  Returns the day of a week (an integer between 1 and 7) from SQL date. Equivalent to EXTRACT(DOW FROM date). E.g., DAYOFWEEK(DATE ‘1994-09-27’) returns 3.

**Note**: Sunday is taken as the first day of the week.

> MOD(numeric1, numeric2): Returns the remainder (modulus) of numeric1 divided by numeric2. The result is negative only if numeric1 is negative.

* When `Day = Sunday`, `MOD(DAYOFWEEK({arg}) + 5, 7) = MOD(1 + 5, 7) = 6`.
* When `Day = Monday`, `MOD(DAYOFWEEK({arg}) + 5, 7) = MOD(2 + 5, 7) = 0`.
...
* When `Day = Friday`, `MOD(DAYOFWEEK({arg}) + 5, 7) = MOD(6 + 5, 7) = 4`.
* When `Day = Saturday`, `MOD(DAYOFWEEK({arg}) + 5, 7) = MOD(6 + 6, 7) = 5`.

So, adding `5` works as expected.